### PR TITLE
fix(http): improve route matching to fix mac subdomain resolution issue

### PR DESCRIPTION
**Pull request by jedna**
Safari and shell interpreters on MacOS don't implicitly resolve subdomains on localhost (e.g. container_id.localhost). This behavior causes an issue with one of our integration tests. Proposed change updates the test to use localhost?canisterId=X schema instead and updates the router matching logic to allow for such change.

### CODE REVIEW

File: libs/orbit-essentials/src/http.rs [hunk 1]
```diff
@@ -61,3 +61,37 @@ fn cbor_encode(value: &impl Serialize) -> Vec<u8> {
...
+
+pub fn not_found() -> HttpResponse {
+    HttpResponse {
+        status_code: 404,
+        headers: vec![HeaderField("Content-Type".into(), "text/plain".into())],
+        body: "404 Not Found".as_bytes().to_owned(),
+    }
+}
+
+pub fn parse_path(url: &str) -> Option<&str> {
```
Comment (mraszyk) [inline libs/orbit-essentials/src/http.rs:73]: Why are you not using a library for this?

Comment (jedna) [reply libs/orbit-essentials/src/http.rs:73]: We discussed this with @keplervital and concluded it's unnecessary to include a library for such a simple and isolated use case where only a dev environment is affected. WDYT?

Comment (jedna) [reply libs/orbit-essentials/src/http.rs:73]: > are you sure that none of the existing dependencies can deal with that?
Good call, I'm not. I'll investigate.

File: core/control-panel/impl/src/controllers/http.rs [hunk 2]
```diff
@@ -35,19 +35,14 @@ impl HttpController {
     }
 
     async fn router(&self, request: HttpRequest) -> HttpResponse {
-        if request.url == "/metrics" || request.url == "/metrics/" {
-            return self.metrics(request).await;
+        match parse_path(&request.url) {
+            Some(url) => match url {
+                "/metrics" | "/metrics/" => self.metrics(request).await,
+                "/metrics/sd" | "/metrics/sd/" => self.metrics_service_discovery(request).await,
```
Comment (keplervital) [inline core/control-panel/impl/src/controllers/http.rs:41]:
```suggestion
            Some(path) => match path.trim_end_matches('/') {
                "/metrics" => self.metrics(request).await,
                "/metrics/sd" => self.metrics_service_discovery(request).await,
```

File: core/station/impl/src/controllers/http.rs [hunk 3]
```diff
@@ -27,15 +27,13 @@ impl HttpController {
     }
 
     async fn router(&self, request: HttpRequest) -> HttpResponse {
-        if request.url == "/metrics" || request.url == "/metrics/" {
-            return self.metrics(request).await;
+        match parse_path(&request.url) {
+            Some(url) => match url {
+                "/metrics" | "/metrics/" => self.metrics(request).await,
```
Comment (keplervital) [inline core/station/impl/src/controllers/http.rs:32]:
```suggestion
            Some(path) => match path.trim_end_matches('/') {
                "/metrics" => self.metrics(request).await,
```

File: libs/orbit-essentials/src/http.rs [hunk 4]
```diff
@@ -61,3 +61,37 @@ fn cbor_encode(value: &impl Serialize) -> Vec<u8> {
...
+    fn test_parse_path() {
+        assert_eq!(parse_path("/long/path?param"), Some("/long/path"));
+        assert_eq!(parse_path("/path?query=1"), Some("/path"));
+        assert_eq!(parse_path("/path#unwanted"), Some("/path#unwanted"));
+        assert_eq!(parse_path("/"), Some("/"));
+        assert_eq!(parse_path(""), Some(""));
+    }
+
+    #[test]
+    fn test_cbor_encode() {
```
Comment (keplervital) [inline libs/orbit-essentials/src/http.rs:91]: what is this test covering?

Comment (jedna) [reply libs/orbit-essentials/src/http.rs:91]: It's a dummy test for the `cbor_encode` function from this module, which had no test. I was mostly curious about the output, so I left it there. I can remove it if it's useless.

File: libs/orbit-essentials/src/http.rs (hunk 1 above)
Comment (mraszyk) [reply libs/orbit-essentials/src/http.rs:73]: I see, thank you for looking into this!

File: libs/orbit-essentials/src/http.rs (hunk 4 above)
Comment (keplervital) [reply libs/orbit-essentials/src/http.rs:91]: i see, let's keep then, but maybe remove the `println!`

**Review by keplervital — APPROVED** [review]

**Pull request merged**